dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -6620,26 +6473,31 @@ int processWhere(SELECT_LEX &select_lex,
     }
 #endif
 
-    uint32_t failed = buildOuterJoin(gwi, select_lex);
+    uint32_t failed = 0;
 
-    if (failed) return failed;
+    // InfiniDB bug5764 requires outer joins to be appended to the
+    // end of the filter list. This causes outer join filters to
+    // have a higher join id than inner join filters.
+    // TODO MCOL-4680 Figure out why this is the case, and possibly
```

Comment:
I presume b/c this reduces the cardinality of the result set. Moreover if a query has an outer joins we process joins sequentially so if we process outer joins first then we loose extra records from outer side applying inner join predicate to an outer join product.